light_curves/code/gaia_functions.py

```diff
@@ -387,41 +296,11 @@ def Gaia_mk_MultiIndex(coords_list, labels_list, gaia_phot , gaia_epoch_phot , v
                         #this_df_lc_gaia exists
                         this_df_lc = pd.concat([this_df_lc, dfsingle])
 
-            else: # No match to Gaia multi-epoch catalog: use single epoch photometry
-                if verbose > 1: print("No Gaia epoch photometry, append single epoch photometry ")
-
-                for band in ["G","BP","RP"]:
-
-                    # get data
-                    t = Time("2015-09-24T19:40:33.468" , format="isot") # just random date: FIXME: NEED TO GET ACTUAL OBSERVATION TIME!
-                    y = gaia_phot["phot_{}_mean_mag".format(band.lower())][sel]
-                    dy = gaia_phot["phot_{}_mean_mag_error".format(band.lower())][sel]
-
-                    # compute flux and flux error in mJy
-                    y2 = 10**(-0.4*(y - 23.9))/1e3 # in mJy
-                    dy2 = dy / 2.5 * np.log(10) * y2 # in mJy
-
-                    # create single instance
-                    dfsingle = pd.DataFrame(
-                                dict(flux=np.asarray(y2), # in mJy
-                                 err=np.asarray(dy2), # in mJy
-                                 time=t.mjd, # in MJD
-                                 #objectid=gaia_phot["input_object_name"][sel],
-                                 objectid=np.repeat(objectid, len(y)),label=lab,
-                                 band="Gaia {}".format(band.lower())
-                                    )
-                    ).set_index(["objectid", "label", "band", "time"])
-
-                    # add to table
-                    try:
-                        this_df_lc
-                    except NameError:
-                        #this_df_lc doesn't exist (yet)
-                        this_df_lc = dfsingle.copy()
-                    else:
-                        #this_df_lc_gaia exists
-                        this_df_lc = pd.concat([this_df_lc, dfsingle])
```

Comment:
If there is no match to any object in `coords_list`, `this_df_lc` never gets defined, and the `return` statement below throws an error. I suspect this is a problem with multiple archive calls, not just this one. One solution would be to return an empty dataframe, but that seems like a hassle. I wonder if a better solution will be for the archive call to return `None` in that case, and then to update the `MultiIndexDFObject.append` method to ignore `None` values.

This problem does not need to hold up this PR, but perhaps we should open an Issue that points to it.

Reply:
good point, I will open an issue to maybe make this more uniform across all functions.  Heasarc currently returns an empty data frame, so does TESS/Kepler/K2.  Those are the ones that routinely will not find any matches for a small sample.  I'm going to tag you @troyraen in that issue because I don't know how to 'ignore' `none` values